prusa/link/cameras/v4l2_driver.py

```diff
@@ -110,83 +110,84 @@ def read_capabilities(file_descriptor):
 
 def read_info(filename):
     """Reads device specific info needed for device initialization"""
-    file_descriptor = fopen(filename)
-    caps = read_capabilities(file_descriptor)
-    version_tuple = (
-        (caps.version & 0xFF0000) >> 16,
-        (caps.version & 0x00FF00) >> 8,
-        (caps.version & 0x0000FF),
-    )
-    version_str = ".".join(map(str, version_tuple))
-    device_capabilities = caps.capabilities
+    with fopen(filename) as file_descriptor:
+        caps = read_capabilities(file_descriptor)
+        version_tuple = (
+            (caps.version & 0xFF0000) >> 16,
+            (caps.version & 0x00FF00) >> 8,
+            (caps.version & 0x0000FF),
+        )
+        version_str = ".".join(map(str, version_tuple))
+        device_capabilities = caps.capabilities
 
-    formats = []
-    pixel_formats = set()
+        formats = []
+        pixel_formats = set()
+
+        fmt = v4l2.v4l2_fmtdesc()
+        fmt.type = STREAM_TYPE
+        for index in range(128):
+            fmt.index = index
+            try:
+                fcntl.ioctl(file_descriptor, v4l2.VIDIOC_ENUM_FMT, fmt)
+            except OSError as error:
+                if error.errno == errno.EINVAL:
+                    break
+                raise
+            try:
+                pixel_format = fmt.pixelformat
```

Comment:
Tohle by mělo (být přidáno k tomu prvnímu try. Takhle se to nepíše jeden try za druhým, i když ta výjimka vypadne z jiného řádku.